File: Data Structures & Algorithms/baseball-game/submission-2.java
class Solution {
    public int calPoints(String[] operations) {
        int total = 0;
        Stack<Integer> stack = new Stack<>();

        // ["5","-2","4","C","D","9","+","+"]
        for(String op : operations){

            if(op.equals("+")){
                int first = stack.pop();
                int newNum = first + stack.peek();
                stack.push(first);
                stack.push(newNum);
                total += newNum;
            } else if(op.equals("C")){
                total -= stack.pop();
            } else if(op.equals("D")){
                stack.push(stack.peek() * 2);
                total+= stack.peek();
            } else{
                stack.push(Integer.parseInt(op));
                total += stack.peek();
            }
        }

        return total;
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        for op in s:
            if op == "]" and stack[-1] == "[":
                stack.pop()
            elif op == "}" and stack[-1] == "{":
                stack.pop()
            elif op == ")" and stack[-1] == "(":
                stack.pop()
            else:
                stack.append(op)
        
        return not stack